src/sentry/sentry_metrics/kafka_metrics_interface.py

```diff
@@ -0,0 +1,142 @@
+from concurrent.futures import Future
+from datetime import datetime
+from typing import Any, Mapping, Optional, Sequence, Union
+
+from arroyo import Message, Topic
+from arroyo.backends.kafka import KafkaPayload, KafkaProducer, build_kafka_configuration
+from django.conf import settings
+
+from sentry.sentry_metrics.metrics_interface import GenericMetricsBackend
+from sentry.sentry_metrics.use_case_id_registry import UseCaseID
+from sentry.utils import json
+from sentry.utils.kafka_config import get_kafka_producer_cluster_options
+
+
+def build_mri(metric_name: str, type: str, use_case_id: UseCaseID, unit: Optional[str]) -> str:
+    mri_unit = "none" if unit is None else unit
+    return f"{type}:{use_case_id.value}/{metric_name}@{mri_unit}"
+
+
+def set_future(f: Future[Message[KafkaPayload]]) -> Union[Future[None], Future[Any]]:
+    new_future: Union[Future[None], Future[Any]] = Future()
+
+    if f.exception() is not None:
+        new_future.set_exception(f.exception())
+    else:
+        new_future.set_result(None)
+
+    return new_future
+
+
+class KafkaMetricsBackend(GenericMetricsBackend):
+    def __init__(self) -> None:
+
+        self.kafka_topic = settings.KAFKA_INGEST_PERFORMANCE_METRICS
+
+        cluster_name = settings.KAFKA_TOPICS[self.kafka_topic]["cluster"]
+        producer_config = get_kafka_producer_cluster_options(cluster_name)
+        self.producer = KafkaProducer(build_kafka_configuration(default_config=producer_config))
+
+    def counter(
+        self,
+        use_case_id: UseCaseID,
+        org_id: int,
+        project_id: int,
+        metric_name: str,
+        value: Union[int, float],
+        tags: Mapping[str, str],
+        unit: Optional[str] = None,
+        retention_days: Optional[int] = 90,
+    ) -> Optional[Future[None]]:
+
+        """
+        Used for emitting a counter metric for internal use cases only.
```

Comment:
nit: No need to prefix docstrings with "Used for..."
```suggestion
        Emit a counter metric for internal use cases only.
```